### PR TITLE
Closes #1813 - platform stats

### LINKED ISSUE #1813: Implement getPlatformActivityStats getter & tests
Make a getter that returns 
```ts
export interface PlatformActivityStatsResult {
  activeUsers: number; // what defines an active user?

  // OrderEvent table for fill events (eventType == 3) where they are the orderCreator or orderFiller address
  // if multiple fills in the same tx count as one trade then also counting just the unique tradeGroupId from those
  numberOfTrades: number;

  openInterest: BigNumber; // contract call

  // MarketCreated table
  marketsCreated: number;

// from the derived market table
// since that will have the latest volume for any given market
  volume: BigNumber;
 
// disputedMarkets get unique marketIds from the DisputeCrowdsourcerCompleted table
  disputedMarkets: number;
}
```

The getter should take a startdate, enddate, and universe.

### PR DESCRIPTION
**Pull request by brianosaurus**
Implements a getter that returns
```ts
export interface PlatformActivityStatsResult {
  activeUsers: number;

  // OrderEvent table for fill events (eventType == 3) where they are the orderCreator or orderFiller address
  // if multiple fills in the same tx count as one trade then also counting just the unique tradeGroupId from those
  numberOfTrades: number;
  openInterest: BigNumber;

  // MarketCreated logs
  marketsCreated: number;

  volume: BigNumber;
  amountStaked: BigNumber;
  disputedMarkets: number;
}
```

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🎇

 This link will always point to this PR's latest changes: https://peek.run/jvgh4v5

**Comment by robert-e-davidson3** [comment]
added to https://github.com/AugurProject/augur/pull/2778 for easier merging into master

File: packages/augur-sdk/src/state/getter/Platform.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    ...await getField<InitialReporterRedeemedLog>('initialReporter', db.findInitialReporterRedeemedLogs.bind(db)),
+    ...await getField<InitialReportSubmittedLog>('reporter', db.findInitialReportSubmittedLogs.bind(db)),
+    ...await getField<MarketCreatedLog>('marketCreator', db.findMarketCreatedLogs.bind(db)),
+    // Order logs are compressed so we parse them first. Our db helper methods make this easy.
+    ...await getField<ParsedOrderEventLog>('orderCreator', db.findOrderCreatedLogs.bind(db)),
+    ...await getField<ParsedOrderEventLog>('orderCreator', db.findOrderCanceledLogs.bind(db)),
+    ...await getField<ParsedOrderEventLog>('orderFiller', db.findOrderFilledLogs.bind(db)),
+    ...await getField<ParticipationTokensRedeemedLog>('account', db.findParticipationTokensRedeemedLogs.bind(db)),
+    ...await getField<TradingProceedsClaimedLog>('sender', db.findTradingProceedsClaimedLogs.bind(db)),
+    ...await getField<TradingProceedsClaimedLog>('sender', db.findTradingProceedsClaimedLogs.bind(db)),
```
Comment (justinbarry) [inline packages/augur-sdk/src/state/getter/Platform.ts:97]: Duplicate?

File: packages/augur-sdk/src/state/getter/Platform.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+async function getActiveUsers(
+  universe: string,
+  startTime: number,
+  endTime: number,
+  db: DB
+): Promise<number> {
+  const getField = makeGetField(universe, startTime, endTime);
+  const users: Address[] = [
+    ...await getField<CompleteSetsPurchasedLog>('account', db.findCompleteSetsPurchasedLogs.bind(db)),
```
Comment (justinbarry) [inline packages/augur-sdk/src/state/getter/Platform.ts:84]: You *could* change the function declaration in db vs calling bind on these. If we are going to build many of these *lense* functions that might be the way to go.

File: packages/augur-sdk/src/state/getter/Platform.ts (hunk 1 above)
Comment (robert-e-davidson3) [reply packages/augur-sdk/src/state/getter/Platform.ts:97]: yup! good catch

**Review by justinbarry — APPROVED** [review]
LGTM

**Pull request closed without merging**